dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/CompanyApi.kt

```diff
@@ -181,9 +184,63 @@ interface CompanyApi {
         value = ["/{companyId}"],
         produces = ["application/json"],
     )
-    @PreAuthorize("hasRole('ROLE_USER') or @CompanyManager.isCompanyPublic(#companyId)")
+    @PreAuthorize("hasRole('ROLE_USER') or @CompanyQueryManager.isCompanyPublic(#companyId)")
     fun getCompanyById(@PathVariable("companyId") companyId: String): ResponseEntity<StoredCompany>
 
+    /**
+     * A method to update company informtion for one specific company identified by its company Id
+     * @param companyId identifier of the company in dataland
+     * @param companyInformation includes the company information
+     * @return updated information about the company
+     */
+    @Operation(
+        summary = "Update company information selectively",
+        description = "Changed elements of a company information behind the given company Id is updated.",
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Successfully updated company information."),
+        ],
+    )
+    @PatchMapping(
+        value = ["/{companyId}"],
+        consumes = ["application/json"],
+        produces = ["application/json"],
+    )
+    @PreAuthorize("hasRole('ROLE_UPLOADER')")
+    fun patchCompanyById(
+        @PathVariable("companyId") companyId: String,
+        @Valid @RequestBody
+        companyInformationPatch: CompanyInformationPatch,
+    ): ResponseEntity<StoredCompany>
+
+    /**
+     * A method to update company informtion entirely
+     * @param companyId identifier of the company in dataland
+     * @param companyInformation includes the company information
+     * @return updated information about the company
+     */
+    @Operation(
+        summary = "Update company information entirely",
+        description = "all elements of a company information behind the given company Id is updated.",
```

Comment:
Replace all company information of the company associated with the given company Id